### PR TITLE
Fix Paginate so it does not require currentPage

### LINKED ISSUE #4507: AMO is returning a server error after any search related action
STR:

1. Go to AMO-dev homepage and perform a search

Actual result:
A server error is received after any search action

Expected results:
Search results are listed

Notes:
- reproduces also  for 'See more" links and  Collections listed on homepage

server error

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #4507

### CODE REVIEW

File: src/core/components/Paginate/index.js [hunk 1]
```diff
@@ -74,9 +75,6 @@ export class PaginateBase extends React.Component {
     if (count === undefined) {
       throw new Error('The count property cannot be undefined');
     }
-    if (currentPage === undefined) {
```
Comment (tofumatt) [inline src/core/components/Paginate/index.js]: Shouldn't the `defaultProps` setting mean this doesn't happen? I forget how `defaultProps` work exactly...

Comment (bobsilverberg) [reply src/core/components/Paginate/index.js]: Correct, which is why I removed the code. `currentPage` will never be undefined, so this `if` statement is superfluous.

Comment (tofumatt) [reply src/core/components/Paginate/index.js]: I'm wondering what would happen if currentPage was null or 0 due to an error in a parent component. Would be nice to have an error thrown if that will break things.

Comment (bobsilverberg) [reply src/core/components/Paginate/index.js]: Ah, good point. I'll check that and if it's an issue I'll issue a fix.

Comment (bobsilverberg) [reply src/core/components/Paginate/index.js]: Guess what? I totally broke pagination with this fix! I'll open a new issue and hopefully it's something I can fix fairly quickly.

Comment (tofumatt) [reply src/core/components/Paginate/index.js]: That's weird, I got pagination working locally when I tried this...

**Pull request merged**